lib/format_parser.rb

```diff
@@ -58,7 +73,22 @@ def self.parse(io)
         # and examine the file more closely.
       end
     end
-    nil # Nothing matched
+    # Return the array of results if something matched or nil otherwise.
```

Comment:
Returning an Array is a good thing here indeed, the simplest conceptually and has a predictable API. It _might_ have merit to introduce some flag (see up the code) to differentiate between "I want the first result whatever it might be" and "I want all the results and I want to make the decisions myself", which could indeed use either an early return within a map (see below)

Reply:
+1 for what Julik said, and also - maybe a silly question but would it make sense to return an empty array here if nothing matches?

Reply:
+2, I wanted to modify it but I wasn't sure if there was some code depending on a nil to trigger some errors or whatnot!